acceptance/Rakefile

```diff
@@ -120,143 +107,60 @@ EOS
-        }
-      else
-        puts "* Couldn't match #{node_info[:node_label]} in pre_suite-run.log"
-      end
-    end
-    pp preserved_config_hash
-
-    File.open('log/latest/preserved_config.yaml', 'w') do |config_file|
-      YAML.dump(preserved_config_hash, config_file)
-    end
-  end
-rescue Errno::ENOENT => e
-  puts "Warning: Couldn't generate preserved_config.yaml #{e}"
-end
+  args = ["--options-file", options_file, hosts_opt(options[:hosts]), tests_opt,
+          *overriding_options.split(' ')].compact
 
-def list_preserved_configurations(secs_ago = ONE_DAY_IN_SECS)
-  preserved = {}
-  Dir.glob('log/*_*').each do |dir|
-    preserved_config_path = "#{dir}/preserved_config.yaml"
-    yesterday = Time.now - secs_ago.to_i
-    if preserved_config = File.exists?(preserved_config_path)
-      directory = File.new(dir)
-      if directory.ctime > yesterday
-        hosts = []
-        preserved_config = YAML.load_file(preserved_config_path).to_hash
-        preserved_config['HOSTS'].each do |hostname,values|
-          hosts << "#{hostname}: #{values['platform']}, #{values['roles']}"
-        end
-        preserved[hosts] = directory.to_path
-      end
-    end
-  end
-  preserved.map { |k,v| [v,k] }.sort { |a,b| a[0] <=> b[0] }.reverse
+  sh("beaker", *args)
 end
 
-def list_preserved_hosts(secs_ago = ONE_DAY_IN_SECS)
+def preserved_hosts(secs_ago = ONE_DAY_IN_SECS)
   hosts = Set.new
-  Dir.glob('log/**/pre*suite*run.log').each do |log|
-    yesterday = Time.now - secs_ago.to_i
-    File.open(log, 'r') do |file|
-      if file.ctime > yesterday
-        file.each_line do |line|
-          matchdata = /^(\w+)(?:\.[\w.]+)? \(.*?\) \d\d:\d\d:\d\d\$/.match(line.encode!('UTF-8', 'UTF-8', :invalid => :replace))
-          hosts.add(matchdata[1]) if matchdata
+  # if secs_ago = 0, just use latest log
+  do_only_log_latest = secs_ago == 0
+  dirs = do_only_log_latest ? Dir.glob('log/latest') : Dir.glob('log/*')
+  yesterday = do_only_log_latest ? Time.now - ONE_DAY_IN_SECS : Time.now - secs_ago.to_i
+  dirs.each do |dir|
+    if dir !~ /^log\/latest$/ || do_only_log_latest
+    hosts_dirs = do_only_log_latest ? Dir.glob(dir) : Dir.glob("#{dir}/*")
+      hosts_dirs.each do |date_dir|
+        preserved_config_path = "#{date_dir}/#{SUTS_FILENAME}"
+        if preserved_config = File.exists?(preserved_config_path)
+          File.open(preserved_config_path, 'r') do |file|
+            if file.ctime > yesterday
+              file.each_line do |line|
+                # match every line without [-]
+                matchdata = /^((?!\[-\]).)*$/.match(line)
+                hosts.add(line) if matchdata
+              end
+            end
+          end
         end
       end
     end
   end
   hosts
 end
 
-def release_hosts(hosts = nil, secs_ago = ONE_DAY_IN_SECS)
-  secs_ago ||= ONE_DAY_IN_SECS
-  hosts ||= list_preserved_hosts(secs_ago)
+def vmpooler_token
+  `grep --extended --only-matching 'vmpooler_token: .*' ${HOME}/.fog`.split(' ')[1]
```

Comment:
For some strange reason, I have a commented out vmpooler token in my fog file along with the real one. So this grep matches both, even though only one is active. Perhaps better to `YAML.load(File_expand_path("~/.fog"))` which will ignore comments and not rely on `grep --extended`?